client/src/services/fileOperationManager.ts

```diff
@@ -462,7 +466,10 @@ class FileOperationManager {
           return;
         }
         if (!copied) {
-          await deleteFile(data.workspaceHandle, dstPath);
+          // We don't want to delete anything if count is 10 since no file has been created
+          if (filenameCount < 10) {
```

Comment:
Condition is a bit weird, I wonder if it would not be safer to do `if (!Path.areSame(srcPath, dstPath))` instead. This way we're covered.

The `Path.areSame` function is dumb, it would benefit from calling `Path.normalize()` on both paths before doing the comparaison, which means it should become async. It can be done either in this PR or in another one.